src/serde/writer.rs

```diff
@@ -49,6 +49,7 @@ impl fmt::Display for SerializeError {
         match *self {
             SerializeError::IoError(ref err) => write!(f, "IoError: {}", err),
             SerializeError::Custom(ref s) => write!(f, "Custom Error {}", s),
+            SerializeError::InfiniteSequence => write!(f, "InfiniteSequence"),
```

Comment:
Please make this a human-readable error message. This is the Display impl not Debug.